test/integration/sonobuoy_integration_test.go

```diff
@@ -224,6 +224,32 @@ func TestNoDoneFile(t *testing.T) {
 	tb = saveToArtifacts(t, tb)
 }
 
+// TestDoneFileDirectory runs a plugin which reports the done file as the results directory itself and not the
+// files or a tarball to ensure that processing works and all results are considered.
+func TestDoneFileDirectory(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	ns, cleanup := getNamespace(t)
+	defer cleanup(false)
+
+	args := fmt.Sprintf("run --image-pull-policy IfNotPresent --wait -p testImage/yaml/job-manual-directory.yaml -n %v", ns)
+	mustRunSonobuoyCommandWithContext(ctx, t, ns, args)
+	tb := mustDownloadTarball(ctx, t, ns)
+	tb = saveToArtifacts(t, tb)
+
+	// Retrieve the sonobuoy results file from the tarball
+	resultsArgs := fmt.Sprintf("results %v --plugin %v --mode dump", tb, "job-manual-directory")
+	resultsYaml := mustRunSonobuoyCommandWithContext(ctx, t, ns, resultsArgs)
+	var resultItem results.Item
+	yaml.Unmarshal(resultsYaml.Bytes(), &resultItem)
```

Comment:
May be good to handle an error here - yaml.Unmarshal returns err

Reply:
Excellent catch, yes we definitely want to catch the error. If there was an error the result would be wrong too but for something like this we want to be explicit about it. 👍